processing/src/main/java/org/apache/druid/query/filter/JavaScriptDimFilter.java

```diff
@@ -327,7 +335,8 @@ public boolean applyInContext(Context cx, Object input)
       if (extractionFn != null) {
         input = extractionFn.apply(input);
       }
-      return Context.toBoolean(fnApply.call(cx, scope, scope, new Object[]{input}));
+      Object fnResult = fnApply.call(cx, scope, scope, new Object[]{input});
+      return Context.toBoolean(boolConvApply.call(cx, scope, scope, new Object[] {fnResult}));
```

Comment:
Instead of evaluating another script to get the final result, how about get value of  `java.lang.Boolean` on `NativeBoolean` object directly like this ?

```
(Boolean)((ScriptableObject) fnResult).getDefaultValue(ScriptRuntime.BooleanClass)
```

Reply:
Thanks, it works. I change to use this way, and add more unit test for other return type.